### ISSUE TITLE
No compilation error shown, groovyc does

### ISSUE THREAD

**Issue by mauromol** [groovy/groovy-eclipse#1345]
Consider an enum:

```java
package test;
public enum MyEnum {
  A, B;
}
```

and a Groovy class:

```groovy
package test
import groovy.transform.CompileStatic

@CompileStatic
class EnumTest {

	MyEnum[] foo() {
		return MyEnum.values().findAll()
	}
}
```

Greclipse does not emit any compilation error, while groovyc does:

```
/home/mauro/workspace/TestGroovyGradle2/src/main/groovy/test/EnumTest.groovy: 9: [Static type checking] - Cannot return value of type java.util.Collection <test.MyEnum> on method returning type test.MyEnum[]
 @ line 9, column 10.
   		return MyEnum.values().findAll()
            ^

1 error
```

Using Greclipse 4.5.0.v202201182034-e2112 with Groovy 2.5 compiler.

**Comment by eric-milles** [comment]
Groovy-Eclipse has bug fixes in advance of Groovy.  Here are some examples: https://github.com/groovy/groovy-eclipse/wiki/4.4.0-Release-Notes#compiler-fixes

**Comment by mauromol** [comment]
I see, but unless these bug fixes are backported to the Groovy compiler versions bundled with Greclipse, this is a problem, because what you see at development time is not what you get when you try to deploy your application (with Gradle, in my case)... Today it happened exactly this, that code apparently correct failed to build.

**Comment by eric-milles** [comment]
I have been merging commits into the 2.5 branch; here is how it looks so far: https://issues.apache.org/jira/browse/GROOVY-10483?jql=project%20%3D%20GROOVY%20AND%20fixVersion%20%3D%202.5.16%20ORDER%20BY%20created%20DESC

You're welcome to join the effort or offer support in some other way -- maybe finding specific tickets that should be prioritized above others.

**Comment by mauromol** [comment]
I don't know which commit should address this (and hence needs a backport), however this is still an issue with Greclipse 4.5.0.v202203070924-e2112 and the newly released Groovy 2.5.16 (I mean: the above case compiles fine with Greclipse, but it does not compile with Groovy 2.5.16 through Gradle).

Today I discovered another static type checker regression in 2.5.16 (compared to 2.5.15) which is once again not shown by Greclipse (I mean: the code compiles fine in Greclipse, while compilation fails in 2.5.16 through Gradle). I'm going to open a JIRA ticket for this in Groovy issue tracker, but let me say that this is quite frustrating: almost on every Groovy 2.5.x (i.e.: old-old-stable now) minor upgrade I have to change something "cosmetic" in my code because static compilation breaks. Having Greclipse behave differently from groovyc does not help at all and causes failures to be intercepted late in the code-build-deploy cycle.

**Comment by eric-milles** [comment]
I did not get this specific issue -- STC collection to array assignment/conversion -- fixed in 2.5.16.  I made quite an effort to backport as much as I reasonably could -- there are 108 tickets associated with 2.5.16.  I tried to balance risk versus reward.  STC was not a primary focus since changes there tend to have the affect you describe, namely casts or missing generics or some small adjustment is required.
